### PR TITLE
[ENH] MNE Scan: isotrak extended header and HPI fit without loading from file

### PR DESCRIPTION
**Pull request by gabrielbmotta**
- Parse fieldtrip extended header id 9 (FT_CHUNK_NEUROMAG_ISOTRAK)
- Add parsed digitizer data to Measurement RTMSA in FTBuffer and FiffSimulator
- Add digitizer data to HPI result
- Align head in 3d view in mne scan without loading digitizer data from small recording of measurement.
- Change default HPI 3d view helmet.

### CODE REVIEW

File: applications/mne_scan/libs/scMeas/realtimemultisamplearray.h [hunk 1]
```diff
@@ -45,6 +45,7 @@
 #include "realtimesamplearraychinfo.h"
 
 #include <fiff/fiff_info.h>
+#include <fiff/c/fiff_digitizer_data.h>
```
Comment (LorenzE) [inline applications/mne_scan/libs/scMeas/realtimemultisamplearray.h:48]: Could we maybe remove these two includes and forward declare them below?

File: applications/mne_scan/plugins/ftbuffer/ftbuffer.cpp [hunk 2]
```diff
@@ -273,6 +273,27 @@ bool FtBuffer::setupRTMSA(FIFFLIB::FiffInfo FiffInfo)
 
 //=============================================================================================================
 
+bool FtBuffer::setupRTMSA(MetaData metadata)
```
Comment (LorenzE) [inline applications/mne_scan/plugins/ftbuffer/ftbuffer.cpp:276]: Can this be a const ref?

**Review by LorenzE — COMMENTED** [review]
Just some (early) minor comments, before I saw this is still WIP. Will continue reviewing once this is supposed to be merged.

File: applications/mne_scan/plugins/ftbuffer/ftbuffer.cpp (hunk 2 above)
Comment (juangpc) [reply applications/mne_scan/plugins/ftbuffer/ftbuffer.cpp:276]: agreed

File: applications/mne_scan/libs/scDisp/realtime3dwidget.cpp [hunk 3]
```diff
@@ -368,7 +362,53 @@ void RealTime3DWidget::alignFiducials(QSharedPointer<FIFFLIB::FiffDigitizerData>
     m_pFiffDigitizerData = pDigData;
 
     if (m_pFiffDigitizerData){
+        MneMshDisplaySurfaceSet* pMneMshDisplaySurfaceSet = new MneMshDisplaySurfaceSet();
```
Comment (juangpc) [inline applications/mne_scan/libs/scDisp/realtime3dwidget.cpp:365]: I know it is hard to keep an eye on this while concentrated on making the thing working but, could we use smart pointers? If we build the habit, it should become less hard.

File: applications/mne_scan/libs/scDisp/realtime3dwidget.cpp [hunk 4]
```diff
@@ -368,7 +362,53 @@ void RealTime3DWidget::alignFiducials(QSharedPointer<FIFFLIB::FiffDigitizerData>
...
+
+        // align and scale average head (now in head space)
+        QList<QStandardItem*> itemList = m_pBemHeadAvr->findChildren(Data3DTreeModelItemTypes::BemSurfaceItem);
+        for(int j = 0; j < itemList.size(); ++j) {
+            if(BemSurfaceTreeItem* pBemItem = dynamic_cast<BemSurfaceTreeItem*>(itemList.at(j))) {
+                pBemItem->setTransform(m_tAlignment);
+            }
+        }
 
+        delete pMneMshDisplaySurfaceSet;
```
Comment (juangpc) [inline applications/mne_scan/libs/scDisp/realtime3dwidget.cpp:411]: see previous comment

File: examples/ex_fiff_sniffer/main.cpp [hunk 5]
```diff
@@ -12,6 +12,17 @@
 #include <fiff/fiff_stream.h>
 #include <fiff/c/fiff_digitizer_data.h>
 
+////////////////////////
+
+#include <connectivity/network/network.h>
```
Comment (juangpc) [inline examples/ex_fiff_sniffer/main.cpp:17]: Connectivity dependency?

File: applications/mne_scan/libs/scDisp/realtime3dwidget.cpp (hunk 3 above)
Comment (gabrielbmotta) [reply applications/mne_scan/libs/scDisp/realtime3dwidget.cpp:365]: This is currently just a copy/paste of the previous allignFiducials() function, but without the requirement of reading from file. If it works(when we test on the neuromag machine), next step is to refactor the calculations to a new function that both the from-file and not-from-file versions can call. In there i think it makes sense to change it.

File: examples/ex_fiff_sniffer/main.cpp (hunk 5 above)
Comment (gabrielbmotta) [reply examples/ex_fiff_sniffer/main.cpp:17]: Sorry I am in the process of throwing a bunch of code in this example's main() to test various points of dealing with the digitizer data without reading from file. These changes are not intended to be merged.

Comment (gabrielbmotta) [reply examples/ex_fiff_sniffer/main.cpp:17]: That being said, this is testing the part of the hpi display code that needed the file, and it works with data found in the isotrak file, so this is good news.

Comment (juangpc) [reply examples/ex_fiff_sniffer/main.cpp:17]: :)

File: applications/mne_scan/libs/scDisp/realtime3dwidget.cpp (hunk 3 above)
Comment (juangpc) [reply applications/mne_scan/libs/scDisp/realtime3dwidget.cpp:365]: great

**Review by juangpc — DISMISSED** [review]

**Review by juangpc — APPROVED** [review]

**Pull request merged**